internal/mode/static/state/graph/gateway_listener.go

```diff
@@ -359,22 +361,33 @@ func createHTTPSListenerValidator() listenerValidator {
 	}
 }
 
-func createPortConflictResolver() listenerConflictResolver {
+func createPortConflictResolver(protectedPorts map[int32]string) listenerConflictResolver {
 	conflictedPorts := make(map[v1beta1.PortNumber]bool)
 	portProtocolOwner := make(map[v1beta1.PortNumber]v1beta1.ProtocolType)
 	listenersByPort := make(map[v1beta1.PortNumber][]*Listener)
 
-	format := "Multiple listeners for the same port %d specify incompatible protocols; " +
+	protcolConflictFormat := "Multiple listeners for the same port %d specify incompatible protocols; " +
 		"ensure only one protocol per port"
+	protectedPortsConflictFormat := "port: Invalid value: %d: port is already in use as %v"
 
 	return func(l *Listener) {
 		port := l.Source.Port
 
+		// if port is in the protected ports list, set current listener as invalid
```

Comment:
I suggest separating this port validation from createPortConflictResolver(), because createPortConflictResolver deals with conflicts among listener. We can create another function and add to `listenerConfigurator.validators` to reject listeners with protected ports.